pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -599,6 +602,16 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 			if found {
 				labels["migration"] = migration
 			}
+
+			_, err = c.kubeClient.Discovery().ServerResourcesForGroupVersion("route.openshift.io/v1")
+			if err == nil {
+				// Openshift
+				annotations["service.beta.openshift.io/inject-cabundle"] = "true"
+			} else {
+				// K8s
+				annotations["cert-manager.io/inject-ca-from"] = fmt.Sprintf("%s/%s", populatorNamespace, "populator-certs")
+			}
```

Comment:
we can rather propagate the indication of openshift vs. k8s from the operator to [the deployment](https://github.com/kubev2v/forklift/blob/main/operator/roles/forkliftcontroller/templates/populator/deployment-populator-controller.yml.j2), no?

Reply:
I've decided to not use this, the certificate is short lived anyway and we won't have this in CDI anyway

Reply:
yeah, that solution doesn't fit the case in which the populator is triggered by CDI
so why did we look into a secure endpoint on the first place?

Reply:
>so why did we look into a secure endpoint on the first place?

it was late 😄

Reply:
yeah, and rainy...